RIOT-OS/RIOT
# gnrc_sixlowpan_frag: move public rbuf functions to their own module

### Contribution description
While implementing [selective fragment recovery](https://tools.ietf.org/html/draft-ietf-6lo-fragment-recovery-05) I ran into the problem of the reassembly buffer being bound to tight to classic reassembly, which makes it basically impossible to use without it. As a first preparation step to factor it out, I moved all its public functions to a new sub-module `gnrc_sixlowpan_frag_rb`. This should simplify the review greatly, as only move stuff around for now (which you can confirm if you activate [diff.colorMoved](https://git-scm.com/docs/git-config#Documentation/git-config.txt-diffcolorMoved) in your git-config.



### Testing procedure
Verify that apart from some header changes and module definition stuff I only move stuff around (I'm assuming the RIOT-OS/RIOT remote is called `upstream` on your system)

```sh
git fetch upstream
git diff --color-moved $(git merge-base upstream/master HEAD)
```

`tests/gnrc_sixlowpan_frag` should still work:

```sh
BOARD="<pick one>" make -C tests/gnrc_sixlowpan_frag flash test
```

All applications involving 6LoWPAN should still build (see Murdock)




### Issues/PRs references
None so far
